### PR TITLE
STY: Remove relative imports, wildcard imports, and convenience methods

### PR DESCRIPTION
**Pull request by jklenzing**
# Description

Addresses #150 

- Removes the wildcard imports from the various `__init__` files. 
- Specifies absolute paths for imports
- Removes convenience methods from datetime and pandas now that pysatCDF is compliant

## Type of change

Please delete options that are not relevant.

- Style changes

# How Has This Been Tested?

Tested that instruments load properly in ipython.  Travis CI tests.

# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``master``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes

### CODE REVIEW

File: pysat/instruments/__init__.py [hunk 1]
```diff
@@ -16,4 +16,5 @@
            'pysat_testing', 'pysat_testing_xarray', 'pysat_testing2d',
            'pysat_testing2d_xarray', 'pysat_testmodel']
 
-from . import *
+for inst in __all__:
+    exec("from pysat.instruments import {x}".format(x=inst))
```
Comment (aburrell) [inline pysat/instruments/__init__.py:20]: Just curious, why this change?

Comment (jklenzing) [reply pysat/instruments/__init__.py:20]: Better maintainability.  I don't want to write out the full list twice in case of typos.  Particularly with the instrument lists changing down the line.  Open to suggestions for cleaner implementation / better style.

Comment (jklenzing) [reply pysat/instruments/__init__.py:20]: I also thought that `import *` was one of the things to eliminate based on the issue.

Comment (aburrell) [reply pysat/instruments/__init__.py:20]: The `exec` caught my attention, it's a shame it can't just be `from pysat.instruments import inst`.  But I agree with all the things you said and am not opposed to the `exec` as it is cleaner than the other options I can think of.

Comment (jklenzing) [reply pysat/instruments/__init__.py:20]: OK.  I'll keep thinking about this as well, but will merge these in.
